### ISSUE TITLE
Bug: binary input files don't work on Python 3.x

### ISSUE THREAD

**Issue by cehbrecht** [geopython/pywps#301]
# Description

I've started a PyWPS service on Python 3.6 and run a process with netCDF (binary) input file. 
I got the following exception, because the current implementation for Python 3.x does not handle binary files (only text):

```
Exception: code: 400, locator: File reference error: 'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte, description: 
Traceback (most recent call last):
  File "/Users/pingu/sandbox/birdhouse/pywps/pywps/app/Service.py", line 409, in href_handler
    (reference_file, reference_file_data) = _openurl(datain)
  File "/Users/pingu/sandbox/birdhouse/pywps/pywps/app/Service.py", line 661, in _openurl
    reference_file_data = reference_file.read().decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte
```

# Environment
- Python version:
Python 3.x
- PyWPS version:
4.x, develop branch

# Steps to Reproduce

Run PyWPS on Python 3.6 with a binary file as complex input (image/png). 

# Additional Information

Already working on a fix using requests:

https://github.com/bird-house/pywps/tree/dev-href-handler

**Comment by jorgejesus** [comment]
:+1:  for use requests and reduce code. Was this tested in python2.7 ?

**Comment by cehbrecht** [comment]
@jorgejesus 
I've tested the fix with requests on Python 2.7 and Python 3.6. requests handles the differences for us.

**Comment by cehbrecht** [comment]
Fixed by PR #302.
